test/simulator/timestamp/call_log_manager/call_log_interface.cpp

```diff
@@ -0,0 +1,49 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <iostream>
+#include <memory>
+
+#include "call_log_manager.h"
+
+int
+main(int argc, char *argv[])
+{
+    /* Return if call log file was not passed. */
+    if (argc != 2) {
+        std::cout << "call_log_interface: missing call log file path" << std::endl;
+        return (-1);
+    }
+
+    std::string call_log_file = argv[1];
+
+    std::unique_ptr<call_log_manager> cl_manager =
+      std::make_unique<call_log_manager>(call_log_file);
```

Comment:
Some places are great for using `auto` and I think this is one of them. Eg.
`auto cl_manager = std::make_unique<call_log_manager>(call_log_file);`

Reply:
Updated in [06d89c5](https://github.com/wiredtiger/wiredtiger/pull/8084/commits/06d89c5806d4e9f5cee8004c7154e2725f3bf0c8)